server/ai_live_video.go

```diff
@@ -33,4 +36,39 @@ func startTricklePublish(url *url.URL, params aiRequestParams) {
 }
 
 func startTrickleSubscribe(url *url.URL, params aiRequestParams) {
+	// subscribe to the outputs and send them into LPMS
+	subscriber := trickle.NewTrickleSubscriber(url.String())
+	r, w, err := os.Pipe()
+	if err != nil {
+		slog.Info("error getting pipe for trickle-ffmpeg", "url", url, "err", err)
+	}
+
+	// read segments from trickle subscription
+	go func() {
+		defer w.Close()
+		for {
+			segment, err := subscriber.Read()
+			if err != nil {
+				// TODO if not EOS then signal a new orchestrator is needed
+				slog.Info("Error reading trickle subscription", "url", url, "err", err)
+				return
+			}
+			defer segment.Body.Close()
+			// TODO send this into ffmpeg
+			io.Copy(w, segment.Body)
+		}
+	}()
+
+	// lpms
+	go func() {
+		ffmpeg.Transcode3(&ffmpeg.TranscodeOptionsIn{
+			Fname: fmt.Sprintf("pipe:%d", r.Fd()),
+		}, []ffmpeg.TranscodeOptions{{
+			// TODO take from params
+			Oname:        "rtmp://localhost/out-stream",
```

Comment:
would this be going into mediaMTX?

Reply:
Yes this is hardcoded into mediamtx for now but we need to take a user configurable output destination. That is where you come in @mjh1 :)